internal/controller/sync.go

```diff
@@ -596,6 +598,12 @@ func (r *RouteController) populateRoute(ctx context.Context, route *routev1.Rout
 	}
 	route.Spec.TLS.Certificate = string(encodedCert)
 
+	encodedCAs, err := utilpki.EncodeX509Chain(certificates[1:])
```

Comment:
this is dangerous imho - since there is no guarantee - or at least i do not see one - that the concrete ca certs start with index 1

Reply:
So we must have 2 functions:
- one to return the certificate linked to the FQDN
- one to return the chain of CA certificates.

Do such functions exist in the `cert-manager` project?